index.js

```diff
@@ -364,7 +365,10 @@ Sharp.prototype.overlayWith = function(overlay, options) {
       throw new Error(' Invalid Value for tile ' + options.tile + 'Only Boolean Values allowed for overlay.tile.');
     }
     
-    if (isInteger(options.gravity) && inRange(options.gravity, 0, 8)) {
+    if(isInteger(options.left) && options.left >= 0 && isInteger(options.top) && options.top >= 0) {
```

Comment:
Given the strictness in the rest of this function, I think if people try to define either `left` or `top` and either are invalid, it should also throw an Error.